torchmetrics/functional/image/uqi.py

```diff
@@ -0,0 +1,245 @@
+
+    mu_pred_sq = output_list[0].pow(2)
+    mu_target_sq = output_list[1].pow(2)
+    mu_pred_target = output_list[0] * output_list[1]
+
+    sigma_pred_sq = output_list[2] - mu_pred_sq
+    sigma_target_sq = output_list[3] - mu_target_sq
+    sigma_pred_target = output_list[4] - mu_pred_target
+
+    upper = 2 * sigma_pred_target
+    lower = sigma_pred_sq + sigma_target_sq
+
+    uqi_idx = ((2 * mu_pred_target) * upper) / ((mu_pred_sq + mu_target_sq) * lower)
+    uqi_idx = uqi_idx[..., pad_h:-pad_h, pad_w:-pad_w]
+
+    return reduce(uqi_idx, reduction)
+
+
+def universal_image_quality_index(
+    preds: Tensor,
+    target: Tensor,
+    kernel_size: Sequence[int] = (11, 11),
+    sigma: Sequence[float] = (1.5, 1.5),
+    reduction: str = "elementwise_mean",
+    data_range: Optional[float] = None,
+) -> Tensor:
+    """Universal Image Quality Index.
+
+    Args:
+        preds: estimated image
+        target: ground truth image
+        kernel_size: size of the gaussian kernel (default: (11, 11))
+        sigma: Standard deviation of the gaussian kernel (default: (1.5, 1.5))
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+
+    Return:
+        Tensor with UQI score
+
+    Raises:
+        TypeError:
+            If ``preds`` and ``target`` don't have the same data type.
+        ValueError:
+            If ``preds`` and ``target`` don't have ``BxCxHxW shape``.
+        ValueError:
+            If the length of ``kernel_size`` or ``sigma`` is not ``2``.
+        ValueError:
+            If one of the elements of ``kernel_size`` is not an ``odd positive number``.
+        ValueError:
+            If one of the elements of ``sigma`` is not a ``positive number``.
+
+    Example:
+        >>> from torchmetrics.functional import universal_image_quality_index
+        >>> preds = torch.rand([16, 1, 16, 16])
+        >>> target = preds * 0.75
+        >>> universal_image_quality_index(preds, target)
+        tensor(0.9216)
+
+    References:
+    [1] Zhou Wang and A. C. Bovik, "A universal image quality index," in IEEE Signal Processing Letters, vol. 9,
+    no. 3, pp. 81-84, March 2002, doi: 10.1109/97.995823.
+    [2] Zhou Wang, A. C. Bovik, H. R. Sheikh and E. P. Simoncelli, "Image quality assessment: from error visibility
+    to structural similarity," in IEEE Transactions on Image Processing, vol. 13, no. 4, pp. 600-612, April 2004,
+    doi: 10.1109/TIP.2003.819861.
+    """
+    preds, target = _uqi_update(preds, target)
+    return _uqi_compute(preds, target, kernel_size, sigma, reduction, data_range)
+
+
+@deprecated(target=universal_image_quality_index, deprecated_in="0.7", remove_in="0.8", stream=_future_warning)
```

Comment:
Since UQI is a new metric being introduced in v0.8, this deprecated function should be removed.